pyquil/_parser/Quil.g4

```diff
@@ -147,10 +163,54 @@ number              : MINUS? ( realN | imaginaryN | I | PI ) ;
 imaginaryN          : realN I ;
 realN               : FLOAT | INT ;
 
+// Analog control
+
+waveformName        : name (DIVIDE name)? ;
+
+defFrame            : DEFFRAME frame ( COLON frameSpec+ )? ;
+frameSpec           : NEWLINE TAB frameAttr COLON ( expression | STRING ) ;
+frameAttr           : SAMPLERATE | INITIALFREQUENCY | DIRECTION | HARDWAREOBJECT | CENTERFREQUENCY;
+
+defWaveform         : DEFWAVEFORM waveformName ( LPAREN param (COMMA param)* RPAREN )? COLON NEWLINE matrix ;
+defCalibration      : DEFCAL name (LPAREN param ( COMMA param )* RPAREN)? qubitOrFormal+ COLON ( NEWLINE TAB instr )* ;
+defMeasCalibration  : DEFCAL MEASURE qubitOrFormal ( name )? COLON ( NEWLINE TAB instr )* ;
+
+pulse               : NONBLOCKING? PULSE frame waveform ;
+capture             : NONBLOCKING? CAPTURE frame waveform addr ;
+rawCapture          : NONBLOCKING? RAWCAPTURE frame expression addr ;
+
+setFrequency        : SETFREQUENCY frame expression ;
+shiftFrequency      : SHIFTFREQUENCY frame expression ;
+setPhase            : SETPHASE frame expression ;
+shiftPhase          : SHIFTPHASE frame expression ;
+swapPhase           : SWAPPHASE frame frame ;
+setScale            : SETSCALE frame expression ;
+
+delay               : DELAY qubitOrFormal+ STRING* expression ;
```

Comment:
This is the only clause (I've found) which requires left recursion, because an expression can match as an identifier (and thus as a qubit variable). Is that intended, and is it worth reconsideration?

Reply:
Will move this issue to rigetti/quil